### DISCUSSION TITLE
cannot add source file

### DISCUSSION THREAD

**Discussion by ranickpatra** [Stellarium/stellarium#670] [Q&A] [answered]
when i try to add new .cpp file it shows compilation error:
because .cpp not generate .o file.
my question is how to add the file

**Comment by alex-w** [comment]
Please give more info about your environment and what you doing. Are you edit cmake rules already?

**Comment by ranickpatra** [comment]
### I,m using Ubuntu 16.04 and it compiles fine without any changes. But when i add a new .cpp and .hpp file with a class "CAR" and call it from main file it shows following error

all my fies are in **stellarium/src** directory

### `Error:`

/home/ranick/Documents/stellarium/src/main.cpp:166: **undefined reference to `CAR::gg()'**
collect2: error: ld returned 1 exit status
src/CMakeFiles/stellarium.dir/build.make:195: recipe for target 'src/stellarium' failed
make[2]: *** [src/stellarium] Error 1
CMakeFiles/Makefile2:430: recipe for target 'src/CMakeFiles/stellarium.dir/all' failed
make[1]: *** [src/CMakeFiles/stellarium.dir/all] Error 2
Makefile:149: recipe for target 'all' failed
make: *** [all] Error 2


### **My files**

### **File aa.hpp**
`class CAR {

    public:

    int gg();

};`

### **File aa.cpp**
`#include "aa.hpp"

int CAR::gg() {
    return 500;
}`

### **main.cpp**

on top I include

`#include "aa.hpp"`

and in main function  I use this 2 line 

CAR l;
l.gg();


this shows above error

**Comment by alex-w** [comment]
You need to update src/CMakeLists.txt also and reconfigure the project

**Comment by ranickpatra** [comment]
How can I do that please help me

**Comment by alex-w** [comment]
Are you using IDE? Are you try to enhance Stellarium through adding plugin or something else?

**Comment by gzotti** [comment, marked as answer]
You edit the CMakeList.txt with any plaintext editor, e.g. in the QtCreator IDE. Try to understand its very simple structure, and then add your source file into the list of source files. Try to follow naming conventions as seen in other source files of the project, give meaningful file, class and variable names etc.

**Comment by ranickpatra** [comment]
@alex-w  I'm trying to extract star's and other object's position for my telescope project.

**Comment by ranickpatra** [comment]
@ranickpatra  Thanks for help I'll try it

**Comment by ranickpatra** [comment]
> You need to update src/CMakeLists.txt also and reconfigure the project

Thanks

**Comment by ranickpatra** [comment]
thanks a lot both of you.

It is working now

**Comment by alex-w** [comment]
> @alex-w I'm trying to extract star's and other object's position for my telescope project.

Why you do not use Telescope Control plugin? For your task (I think you use another Arduino-based mount) you need write a plugin, enhance the Telescope Control plugin or just use LX200/NexStar protocol.
